### PR TITLE
fix render thread tests MAPSAND-337

### PR DESCRIPTION
**Pull request by alexander-kulikovskii**
### Summary of changes

Sometimes our `onSurfaceWithActivityDestroyedAfterSurfaceTest` becames flaky. One of the main idea is that we have some unstable mockk that can affect this tests. So I add `unmockkAll` after each test. Plus bump some of our test dependencies.

Run multiple times locally and they works fine.

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -128,7 +128,7 @@ commands:
     steps:
       - run:
           name: Run JVM unit tests & create jacoco code coverage report
-          no_output_timeout: 30m
+          no_output_timeout: 15m
```
Comment (yunikkk) [inline .circleci/config.yml:131]: Nice, suppose that's unrelated to the flaky tests itself? Maybe make this even smaller to fail faster?

**Comment by yunikkk** [comment]
Do you have some specific ideas on how mockk could be flaky? Not sure I got the idea. 
Also - maybe add a note to the test itself (with reference to this PR), if it won't be solved?

File: .circleci/config.yml (hunk 1 above)
Comment (alexander-kulikovskii) [reply .circleci/config.yml:131]: yes, it's no related to flaky tests. But we have another issues with tests and wait too long before this timeout

**Comment by alexander-kulikovskii** [comment]
> Do you have some specific ideas on how mockk could be flaky? Not sure I got the idea. Also - maybe add a note to the test itself (with reference to this PR), if it won't be solved?

It might be mock + robolectric. Also bump it

**Review by yunikkk — APPROVED** [review]

**Review by ank27 — APPROVED** [review]

**Review by pengdev — APPROVED** [review]

**Pull request merged**